main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -280,6 +280,10 @@ object Simplifier {
 
           case TypeConstructor.Native(clazz) => MonoType.Native(clazz)
 
+          case TypeConstructor.MethodReturnType(name, arity) => throw InternalCompilerException("MethodReturnType shouldn't exist.", tpe.loc)
+
+          case TypeConstructor.StaticMethodReturnType(clazz, name, arity) => throw InternalCompilerException("StaticMethodReturnType shouldn't exist.", tpe.loc)
```

Comment:
By convention, "error cases" go towards the bottom in the pattern match. See below for some of the other pattern matches.

Reply:
I think we should just say "Unexpected type" like the rest.